### PR TITLE
Simplify catalogs and auto-fetching from Zenodo

### PR DESCRIPTION
**Pull request by willirath**
This partly addresses #8 and shows how to just fetch all files associated with a Zenodo record.

ToDo
- ~~What about the mesh archive files? Can we track them with intake and pass them to `pyfesom2`?~~
- [x] Filtering for only those files that are needed.
- ~~Better integration with Intake. Add very simple `pyfesom2` driver? (This could be done in `pyfesom2` directly.)~~
- [x] Test if this also works with the moving DOIs that always point to the latest version on Zenodo.

Remark about the intake integration: At this stage, we could easily stick to not using a catalog, just `curl`ing what we need into a predefined dir, and relying on file-based stuff. But as we will need the catalog functionality later, I strongly suggest, we use it right away to really gain experience with Intake and to iron out things right when they occur.

### CODE REVIEW

**Comment by willirath** [comment]
@koldunovn After we spoke on Tuesday, I lost the implementation I showed you in the meeting, because I forgot to push...

`download_zenodo_files` in <https://github.com/ESM-VFC/esm-vfc-catalogs/blob/b873f6de5ffecd863d2c336ed6122f7dbbb7b275/examples/fetch_based_on_zenodo_doi.md> shows the logic I'm having in mind. We'd ideally start from a relatively clean catalog file (<https://github.com/ESM-VFC/esm-vfc-catalogs/blob/b873f6de5ffecd863d2c336ed6122f7dbbb7b275/catalogs/fesom2_catalog.yaml>) and then fetch data based on just the catalog file and the DOI / Zenodo record ID.

**Comment by willirath** [comment]
See https://github.com/intake/intake/issues/499 for a solution to the tar caching issue: Use zip and then have
```yaml
urlpath: "simplecache::zip://pi/nod2d.out::file://{{env('ESM_VFC_DATA_DIR')}}/FESOM2_PI_MESH/pi.zip"
```

File: catalogs/FESOM2_PI_MESH.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      xarray_kwargs:
+        decode_cf: False
+        combine: 'by_coords'
+
+  # CAUTION: The following is broken with current intake caching!
+  MESH_NOD2D:
+    driver: csv
+    description: 'Node locations of sample FESOM pi mesh'
+    metadata:
+      zenodo_doi: "10.5281/zenodo.3819896"
```
Comment (koldunovn) [inline catalogs/FESOM2_PI_MESH.yaml]: I have uploaded the new version with pi.zip
https://zenodo.org/record/3865567

**Review by koldunovn — COMMENTED** [review]
I just upploaded the new version with zipped mesh: https://zenodo.org/record/3865567
Do you want to try it out? I can also do a PR with changed dio if you wish.

**Comment by willirath** [comment]
> Do you want to try it out? I can also do a PR with changed dio if you wish.

I'll try. But I first need to get a new IP address as Zenodo rate-limited me ... :)

**Comment by willirath** [comment]
Hhhmm. Doesn't seem to work with `zip` either. See: https://github.com/intake/intake/issues/499#issuecomment-636078732

@koldunovn Maybe it's time for a grid-to-netCDF convention?

**Comment by willirath** [comment]
> Hhhmm. Doesn't seem to work with zip either. See: intake/intake#499 (comment)

Works now with `fsspec@master`.

**Comment by willirath** [comment]
I'll merge this so we can update https://github.com/esm-vfc/esm-vfc-workflows.

**Pull request merged**